test/view_component/integration_test.rb

```diff
@@ -558,4 +558,29 @@ def test_renders_a_preview_with_image_path
 
     assert_includes response.body, "images/foo.png"
   end
+
+  def test_sets_the_compiler_mode_in_production_mode
+    old_env = Rails.env
+    begin
+      Rails.env = "production"
```

Comment:
```suggestion
      Rails.env = "production".inquiry
```

`Rails.env` is normally an `ActiveSupport::StringInquirer` - see [here](https://api.rubyonrails.org/classes/ActiveSupport/StringInquirer.html) for more details on that.